docs/src/test/scala/docs/http/scaladsl/HttpClientExampleSpec.scala

```diff
@@ -125,13 +126,27 @@ class HttpClientExampleSpec extends WordSpec with Matchers with CompileOnlySpec
 
         val connectionFlow: Flow[HttpRequest, HttpResponse, Future[Http.OutgoingConnection]] =
           Http().outgoingConnection("akka.io")
-        val responseFuture: Future[HttpResponse] =
-          Source.single(HttpRequest(uri = "/"))
+
+        def dispatchRequest(request: HttpRequest): Future[HttpResponse] =
+          // This is actually a bad idea in general. Even if the `connectionFlow` was instantiated only once above,
+          // a new connection is opened every single time, `runWith` is called. Materialization (the `runWith` call)
+          // and opening up a new connection is slow.
+          //
+          // The `outgoingConnection` API is very low-level. Use it only if you already have a `Source[HttpRequest]`
+          // (other than Source.single) available that you want to use to run requests on a single persistent HTTP
+          // connection.
+          //
+          // Unfortunately, this case is so uncommon, that we couldn't come up with a good example.
+          //
+          // In almost all cases it is better to use the `Http().singleRequest()` API instead.
```

Comment:
Maybe for a while, otherwise, people might find a similar example on SO or so. Hopefully, they also come here and try to find out what's the matter.